### PR TITLE
Allow Python 3.6 jobs on DataflowRunner

### PR DESCRIPTION
**Pull request by charlesccychen**
This change allows submission of Python 3.6 jobs on DataflowRunner.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @aaltay @tvalentyn @markflyhigh

**Comment by charlesccychen** [comment]
Run Python PostCommit

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 1]
```diff
@@ -957,10 +957,10 @@ def _verify_interpreter_version_is_supported(pipeline_options):
   if sys.version_info[0] == 2:
     return
 
-  if sys.version_info[0] == 3 and sys.version_info[1] == 5:
-    if sys.version_info[2] < 3:
+  if sys.version_info[0:2] in [(3, 5), (3, 6)]:
+    if sys.version_info[0:3] < (3, 5, 3):
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Can we remove this warning now?

IIUC, According to @tvalentyn 3.5.0 now works.

Comment (charlesccychen) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: I recall this was a patch for typehints which may be what this was referring to.  @tvalentyn should we keep this recommendation regardless?

Comment (charlesccychen) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: I will send out a separate change if @tvalentyn believes this is no longer necessary.

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: After typehints patch, we are not aware of issues why 3.5.0 might not work. It's a good question whether we want to recommend the user to use the same exact version, or higher version, as we install on Dataflow workers, for example: >= 2.7.13, >= 3.5.6, etc.

**Pull request merged**